src/components/AudioPlayer/SeekButton.tsx

```diff
@@ -36,60 +18,22 @@ type SeekButtonProps = {
   type: SeekButtonType;
   isLoading: boolean;
 };
-const SeekButton = ({ type, isLoading }: SeekButtonProps) => {
-  const { t, lang } = useTranslation('common');
-  const dispatch = useDispatch();
-  const chaptersData = useContext(DataContext);
-  const { highlightedChapter, highlightedVerse } = useSelector(selectHighlightedLocation);
-  const { value: reciterId }: { value: number } = useGetQueryParamOrReduxValue(QueryParam.Reciter);
-  const audioData = useSelector(selectAudioData);
-  const isInRepeatMode = useSelector(selectIsInRepeatMode);
-  const chapterData = useMemo(
-    () => getChapterData(chaptersData, highlightedChapter?.toString()),
-    [chaptersData, highlightedChapter],
-  );
-
-  const { data: chapterAudioData } = useSWRImmutable(
-    reciterId && audioData?.chapterId
-      ? makeChapterAudioDataUrl(reciterId, audioData?.chapterId, true)
-      : null, // only fetch when reciterId and chapterId is available
-    () => getChapterAudioData(reciterId, audioData?.chapterId, true),
-  );
+const SeekButton = ({ type }: SeekButtonProps) => {
+  const audioService = useContext(AudioPlayerMachineContext);
 
-  const verseTimingData = chapterAudioData?.verseTimings || [];
+  const { t } = useTranslation('common');
 
   const onSeek = () => {
     // eslint-disable-next-line i18next/no-literal-string
     logButtonClick(`audio_player_${type}`);
-    if (isInRepeatMode) {
-      // when in repeatMode, finish the repeat progress for current ayah
-      // otherwise the AudioRepeatManager will replay the current Ayah when we set the new timestamp
-      if (type === SeekButtonType.NextAyah) dispatch(finishRepeatEachVerseProgress());
-      else dispatch(resetRepeatEachVerseProgress(lang));
-    }
-    const newVerse = type === SeekButtonType.PrevAyah ? highlightedVerse - 1 : highlightedVerse + 1;
-    const verseKey = makeVerseKey(highlightedChapter, newVerse);
-
-    const selectedVerseTiming = getVerseTimingByVerseKey(verseKey, verseTimingData);
-    triggerSetCurrentTime(selectedVerseTiming.timestampFrom / 1000); // AudioPlayer accept 'seconds' instead of 'ms'
+    audioService.send({ type: type === SeekButtonType.NextAyah ? 'NEXT_AYAH' : 'PREV_AYAH' });
   };
 
-  // disable the button if loading, or chapterAudioData not available, or highlighted verse not available
-  // or currently playing first verse or last verse
-  const isDisabled =
-    isLoading ||
-    !highlightedChapter ||
-    !highlightedVerse ||
-    !chapterAudioData ||
-    (type === SeekButtonType.PrevAyah && highlightedVerse <= 1) ||
-    (type === SeekButtonType.NextAyah && highlightedVerse >= chapterData?.versesCount);
-
   return (
     <Button
       tooltip={type === SeekButtonType.PrevAyah ? t('previous-ayah') : t('next-ayah')}
       variant={ButtonVariant.Ghost}
       shape={ButtonShape.Circle}
-      isDisabled={isDisabled}
```

Comment:
Don't we need to disable the button when we reach either the first or last verse of the current Surah?